### ISSUE TITLE
ajenti wants to use flash ?

### ISSUE THREAD

**Issue by jolly-jump** [linuxmuster/linuxmuster-webui7#131]
Do we **really** need Flash?
whatfor?
does not look good...

![grafik](https://user-images.githubusercontent.com/2824334/79908850-45030d00-841c-11ea-9720-ada0f1d3bfc0.png)

**Comment by kiarn** [comment]
Hello @jolly-jump,

It's weird that firefox propose this, it should use websocket instead. I must investigate, but I never encountered this situation.
So the answer is : we absolutely not need flash.

**Comment by kiarn** [comment]
Hello @jolly-jump,

I'm not able to reproduce this, so I cannot test if my change avoid this behaviour. Can you please test something for me ?

I don't know if you use Ajenti with Python3 or the old version, so I give the instructions for Python3, you may have to adapt : 

1. Comment out the flashsocket use in `/usr/local/lib/python3.6/dist-packages/aj/core.py`. This is the line containing `str('flashsocket'),`, it shloud then be : `#str('flashsocket'),` (only add a # before the s).
2. Restart the Webui : `systemctl restart linuxmuster-webui`,
3. Test in Firefox if the notification about use of flash is still there.

This would be helpful for us, thanks.

**Comment by PLanB2008** [comment]
I saw this issue before, but it disappears after some time

**Comment by kiarn** [comment]
Ok. I have disabled the use of flashsocket in Ajenti : https://github.com/kiarn/ajenti/commit/3b0c49a3fa98880ab3de13c1bead9bd82a26eb9e and I'm pretty sure this is the only dependency to flash, if @jolly-jump or another can confirm it, it would be great :)

**Comment by PLanB2008** [comment]
What was this socket use for? I saw firefox asking for permission but it does not appear on a regular base, at least not on my machine.

**Comment by kiarn** [comment]
This socket is not really used : it's an optional handler listed and proposed when socket.io is started. Ajenti propose a list of transports ( xhr_polling, websocket, flashsocket, ... ), the client browser choose automatically the best one and use it.

See : https://gevent-socketio.readthedocs.io/en/latest/handler.html#module-socketio.handler

**Comment by jolly-jump** [comment]
> I don't know if you use Ajenti with Python3 or the old version, so I give the instructions for Python3, you may have to adapt :

I use the most current linuxmuster-webui debian package now: 1.0.141-1 so I guess it is python3?
 
>     1. Comment out the flashsocket use in `/usr/local/lib/python3.6/dist-packages/aj/core.py`. This is the line containing `str('flashsocket'),`, it shloud then be : `#str('flashsocket'),` (only add a # before the s).
> 
>     2. Restart the Webui : `systemctl restart linuxmuster-webui`,

does not work, still flashing the sign. And more:
I also removed the file `rm -rf /usr/local/lib/python3.6/dist-packages/aj/__pycache__/core.cpython-36.pyc` to be sure.

And now I tried python2.7, changes and:
```
rm /usr/local/lib/python2.7/dist-packages/aj/core.pyc
systemctl restart linuxmuster-webui
```
still flashing... :chicken:

The transport list seems a good choice, but maybe I seem to overlook something?

**Comment by kiarn** [comment]
Hi @jolly-jump 

Thank you for your test !
It's maybe necessary to empty the cache of the browser before testing again.
I tried to test with firefox on many systems, and was not able to reproduce this, it's weird.

Maybe I overlooked something, I must search again ...

**Comment by jolly-jump** [comment]
Hi @kiarn, 

maybe here:
/usr/local/lib/python3.6/dist-packages/ajenti_plugin_core/resources/build/all.vendor.js
at least I can grep a couple of files within the plugin_core directory.
And: hm, I most of the time closed Firefox and opened it again *and* tried it again with a private window.
hm, how to make sure it is not in the cache?

**Comment by jolly-jump** [comment]
Hi @kiarn , 

Bingo!
I have tried a new profile in Firefox: works perfectly using the comment as proposed: No flash in sight.
Checked against: uncommented it again, and the same session reloaded shows the flash.
Now I can't get it out in the second profile either.
A third profile does again not show flash (wenn flashsocket was commented out)

Cheers!

**Comment by kiarn** [comment]
Hi @jolly-jump 

Many thanks for your tests !
So this confirm that the notification is due to the flashsocket transport, and I can close this issue as it is already fixed in ajenti 2.1.35.
